### ISSUE TITLE
cy.clock throws error on second invocation >= 3.1.1

### ISSUE THREAD

**Issue by lauraseidler** [cypress-io/cypress#2850]
### Current behavior:

Calling `cy.clock` a second time within the same test run (but across multiple tests within that) results in the error `TypeError: _global.setTimeout is not a function`. This only appears from 3.1.1 onwards, 3.1.0 is fine.

### Desired behavior:

Being able to call cy.clock multiple times.

### Steps to reproduce:

This is a minimal example that fails for me on 3.1.1 and beyond but passes in 3.1.0
```js
describe('Clock Test', () => {
    it('works the first time', () => {
        cy.clock();
        cy.visit('/');
    });

    it('does not work the second time', () => {
        cy.clock();
        cy.visit('/')
    });
});
```

I assume this might be related to the changes made due to https://github.com/cypress-io/cypress/issues/2570, as the error appears within the `withGlobal` method which calls `_global.setTimeout`.

### Versions

Cypress: >= 3.1.1
Browser: Chrome
OS: MacOS + Debian (in CI)

**Comment by jahredhope** [comment]
I've also been experiencing this issue.
Downgrading to 3.1.0 seems to be a successful work-around.


My **beforeEach** contains
```js
cy.clock(new Date('2018-01-01T00:00:00.000Z').getTime());
```

Which ever test runs first passes, the second test fails during the before step with:

```
TypeError: _global.setTimeout is not a function
```

**Comment by brian-mann** [comment]
Released in [`3.1.3`](https://docs.cypress.io/guides/references/changelog.html#3-1-3).
